libs/data_providers_sdk/src/price_data/fetchers/fetch.rs

```diff
@@ -0,0 +1,35 @@
+use std::time::Instant;
+
+use anyhow::Result;
+use futures::stream::StreamExt;
+use futures::Stream;
+
+use crate::price_data::{traits::prices_fetcher::PairPriceData, types::ProviderPriceData};
+
+pub async fn fetch_all_prices<S>(mut futures_set: S) -> Result<Vec<ProviderPriceData>>
+where
+    S: Stream<Item = (&'static str, Result<PairPriceData>)> + Unpin,
+{
+    let before_fetch = Instant::now();
+    let mut all_fetched_prices: Vec<ProviderPriceData> = Vec::new();
```

Comment:
nit: Maybe don't take the `Vec` allocation time (however small it may be) into account

```suggestion
    let mut all_fetched_prices: Vec<ProviderPriceData> = Vec::new();
    let before_fetch = Instant::now();
```

Reply:
Resolved this one